cmd/ctr/commands/images/export.go

```diff
@@ -69,6 +77,56 @@ When '--all-platforms' is given all images in a manifest list must be available.
 			return errors.New("please provide both an output filename and an image reference to export")
 		}
 
+		client, ctx, cancel, err := commands.NewClient(context)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+
+		var w io.WriteCloser
+		if out == "-" {
+			w = os.Stdout
+		} else {
+			w, err = os.Create(out)
+			if err != nil {
+				return err
+			}
+		}
+		defer w.Close()
+
+		if !context.BoolT("local") {
+			pf, done := ProgressHandler(ctx, os.Stdout)
+			defer done()
+
+			var specified []ocispec.Platform
+			if pss := context.StringSlice("platform"); len(pss) > 0 {
+				for _, ps := range pss {
+					p, err := platforms.Parse(ps)
+					if err != nil {
+						return fmt.Errorf("invalid platform %q: %w", ps, err)
+					}
+					specified = append(specified, p)
+				}
+			}
+
+			err := client.Transfer(ctx,
+				image.NewStore(""), // a dummy image store
```

Comment:
We can create a new type that represents multiple images. The idea is that the source could be anything, it could be a single image from the image store or in this case multiple images from the image store. It could also be sourced from a registry. The `images` shouldn't be used as export options though since that is information about the source.

Reply:
> We can create a new type that represents multiple images

Yeah we could do that, shall we add a new type `MultiImageStore` with a `GetImages` method? like:

```
type MultiImageStore struct {
  images [] string
}

func (s *MultiImageStore) GetImageNames() []string {
  return s.images
}
```

then we could add an interface for it:

```
type ImageNamesGetter interface {
  GetImageNames() []string
}
```

then we could pass it to the method `exportStream` of `localTransferService` to get the image names specified by user.


I am not sure if there is a better way to do this, from my perspective how we get image in `export` is similar to `push`, the only difference is we could export multiple images, and the current implementation of `image.Store` only takes single image into consideration.

I propose we could split the current `image.Store` into a `source image store` and a `destionation image store`, while we could get multiple images from a `source image store`, but could only save only single image to a `destination image store`. Therefore we could reuse the `source image store` in `export` and `push`(we might want to support pushing multiple images later :P) WDYT?

Reply:
We can likely re-use the `ImageStore` and provide an extra names array or extend to names. I think extra names may make sense so that it can implement a `GetImage` but optionally provide more image names. There could be cases in the future where we want to provide additional references on pull as well (such as store by tag@digest, a reference which might be created dynamically on store).

Reply:
I'm playing around with a change to the image store transfer interface and import logic that may make this cleaner.

Reply:
> I'm playing around with a change to the image store transfer interface and import logic that may make this cleaner.

@dmcgowan That's great! I would like to help if there is anything I could do